src/session/backends/local-session-backend.ts

```
@@ -0,0 +1,832 @@
+            this.emit({ type: 'error', message: error.message });
+            settleReject(error);
+          },
+        });
+
+        this.sessionSocket = connection;
+        this.sessionSocketSessionId = expectedSessionId;
+        connection.sendControl({ type: 'attach-init', cols, rows, clientType: 'cli' });
+      } catch (error) {
+        settleReject(toError(error, `Failed to connect to session ${session.name}`));
+      }
+    });
```

Biome (2.3.14)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)